From b7a82a918f0a595a96047f7eef2672fd95d5626c Mon Sep 17 00:00:00 2001
From: Wolfgang Hoschek <whoschek@cloudera.com>
Date: Thu, 11 Dec 2014 14:01:27 -0800
Subject: [PATCH] PARQUET-145 InternalParquetRecordReader.close() should not
 throw an exception if initialization has failed

PARQUET-145 InternalParquetRecordReader.close() should not throw an exception if initialization has failed

Author: Wolfgang Hoschek <whoschek@cloudera.com>

Closes #93 from whoschek/PARQUET-145-3 and squashes the following commits:

52a6acb [Wolfgang Hoschek] PARQUET-145 InternalParquetRecordReader.close() should not throw an exception if initialization has failed
---
 .../main/java/parquet/hadoop/InternalParquetRecordReader.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java b/parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java
index 5a9b019d40..9a0d5d0cef 100644
--- a/parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java
+++ b/parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java
@@ -131,7 +131,9 @@ private void checkRead() throws IOException {
   }
 
   public void close() throws IOException {
-    reader.close();
+    if (reader != null) {
+      reader.close();
+    }
   }
 
   public Void getCurrentKey() throws IOException, InterruptedException {
